### DISCUSSION TITLE
How SPAdes prevents self-loops while breaking sequences into even-sized words?

### DISCUSSION THREAD

**Discussion by apcamargo** [ablab/spades#1353] [Q&A] [answered]
In a [previous discussion](https://github.com/ablab/spades/discussions/1352), it was explained that in SPAdes, the overlaps in the assembly graph have a length of *k*, and sequences are broken down into *(k+1)*-mers. In my assemblies, the overlap length is indeed 55 bp.

However, my understanding is that, for assembly, sequences should be broken into odd-sized *k*-mers to prevent self-loops in the graph, caused by palindromes. This would result in even-sized overlaps. For example, in Minia, the overlaps are *(k-1)* bp long, where *k* is an odd number.

Given that SPAdes uses *(k+1)*-mers, how are self-loops avoided in the assembly process?

**Comment by asl** [comment, marked as answer]
There are two different "flavours" of de Bruijn graph, so-called "edge-centric" and "node-centric". SPAdes employs edge-centric approach. See for example https://www.biostars.org/p/175058/ for some explanation

**Reply by apcamargo** [reply]
Thank you!
